### ISSUE TITLE
Module interface files of dev-dependencies don't need to be installed

### ISSUE THREAD

**Issue by tueda** [fortran-lang/fpm#632]
### Description

If I understand correctly, `dev-dependencies` in the `fpm.toml` file specifies dependencies for development (e.g., tests), and this dependency should not be exposed to dependent projects in any way. But when installing a library by `fpm install`, fpm `0.5.0` seems to copy module interface files of `dev-dependencies` of the library, which should not happen.

**Steps to reproduce the problem:**

```bash
fpm new bug --lib --test
cd bug
sed -i 's/library = false/library = true/' fpm.toml
echo '[dev-dependencies]' >>fpm.toml
echo 'test-drive.git = "https://github.com/fortran-lang/test-drive"' >>fpm.toml
fpm install --prefix $(pwd)
```
Here, the `sed ...` and `echo ...` commands write the following lines in `fpm.toml`:
```toml
[install]
library = true
[dev-dependencies]
test-drive.git = "https://github.com/fortran-lang/test-drive"
```
Then, `fpm install` causes installation of `.mod` files of `dev-dependencies` (`testdrive.mod` and `testdrive_version.mod` in this example):
```
# Update: build/gfortran_E691F5CC75D6381A/bug/libbug.a -> /home/tueda/tmp/20211211/bug/lib
# Update: build/gfortran_E691F5CC75D6381A/bug.mod -> /home/tueda/tmp/20211211/bug/include
# Update: build/gfortran_E691F5CC75D6381A/testdrive.mod -> /home/tueda/tmp/20211211/bug/include
# Update: build/gfortran_E691F5CC75D6381A/testdrive_version.mod -> /home/tueda/tmp/20211211/bug/include
```

### Expected Behaviour

`fpm install` should not install module interface files of `dev-dependencies`.

### Version of fpm

0.5.0

### Platform and Architecture

Ubuntu 20.04.2 (5.10.60.1-microsoft-standard-WSL2), x86_64

### Additional Information

fpm 0.5.0 and gfortran 11.2.0_3 installed via Homebrew.

**Comment by awvwgk** [comment]
Indeed, target specific dependencies and development dependencies shouldn't get installed. The installation step for the modules is implemented here

https://github.com/fortran-lang/fpm/blob/d99ad594f731511413f5bad96f0c0dfe0c2c08aa/src/fpm/cmd/install.f90#L72-L73

by filtering out all modules from the targets:

https://github.com/fortran-lang/fpm/blob/d99ad594f731511413f5bad96f0c0dfe0c2c08aa/src/fpm/cmd/install.f90#L119

The implementation of the filter step can be found here:

https://github.com/fortran-lang/fpm/blob/d99ad594f731511413f5bad96f0c0dfe0c2c08aa/src/fpm_targets.f90#L679-L700

An additional scope information to only select module files of required projects could be passed, but I'm currently not sure whether the targets still know which project they are associated to. I think we have to install all module files from the whole dependency tree of the library, since we include all those sources in the archive at the moment.
